modules/apps/document-library/document-library-preview-pdf/src/main/java/com/liferay/document/library/preview/pdf/internal/portlet/action/EditPDFPreviewMVCActionCommand.java

```diff
@@ -68,9 +68,15 @@ protected void doProcessAction(
 		}
 
 		try {
+			int maxNumberOfPages = ParamUtil.getInteger(
+				actionRequest, "maxNumberOfPages");
+
+			if (maxNumberOfPages < 0) {
+				throw new PDFPreviewException();
```

Comment:
Can you add a simple error message with the failure reason? E.g. "Maximum number of pages must be a non-negative integer".

Reply:
Do you think it's necessary?
The validation will be done on the front-end part, 
This is a fallback (guarding our backs... pun intended) value on the "exceptional" case that skips the front.

Reply:
I think it is useful. It takes almost no effort and gives some context on what the error is about if it shows up later in some log file.